packages/search-in-workspace/src/node/ripgrep-search-in-workspace-server.ts

```diff
@@ -93,47 +98,91 @@ export class RipgrepSearchInWorkspaceServer implements SearchInWorkspaceServer {
+        const args = new Set<string>();
+
+        const appendGlobArgs = (rawPatterns: string[], exclude: boolean) => {
+            rootPaths.forEach(root => {
+                for (const rawPattern of rawPatterns) {
+                    if (rawPattern !== '') {
+                        const globArguments = this.pathToGlobs(rawPattern, root, exclude);
+                        globArguments.forEach(arg => args.add(arg));
+                    }
+                }
+            });
+        };
+
+        args.add('--hidden');
+        args.add('--json');
+
+        args.add(options && options.matchCase ? '--case-sensitive' : '--ignore-case');
         if (options && options.includeIgnored) {
-            args.push('--no-ignore');
+            args.add('--no-ignore');
         }
         if (options && options.maxFileSize) {
-            args.push('--max-filesize=' + options.maxFileSize.trim());
+            args.add('--max-filesize=' + options.maxFileSize.trim());
         } else {
-            args.push('--max-filesize=20M');
+            args.add('--max-filesize=20M');
         }
+
         if (options && options.include) {
-            for (const include of options.include) {
-                if (include !== '') {
-                    args.push('--glob=**/' + include);
-                }
-            }
+            appendGlobArgs(options.include, false);
         }
+
         if (options && options.exclude) {
-            for (const exclude of options.exclude) {
-                if (exclude !== '') {
-                    args.push('--glob=!**/' + exclude);
-                }
-            }
+            appendGlobArgs(options.exclude, true);
         }
+
         if (options && options.useRegExp || options && options.matchWholeWord) {
-            args.push('--regexp');
+            args.add('--regexp');
         } else {
-            args.push('--fixed-strings');
-            args.push('--');
+            args.add('--fixed-strings');
+            args.add('--');
         }
-        return args;
+
+        return Array.from(args);
     }
 
+    protected pathToGlobs(inputPath: string, rootFolder: string, exclude: boolean): string[] {
+        const prefixPattern = (inputPattern: string): string => {
+            const globalCommandArgument = '--glob=';
+            const excludeChar = exclude ? '!' : '';
+            const subDirGlobPattern = '**/';
+
+            const subDirGlobPrefix = inputPattern.startsWith('/') ? '**' : subDirGlobPattern;
+            const updatedPattern = inputPattern.startsWith(subDirGlobPattern) ?
+                inputPattern : `${subDirGlobPrefix}${inputPattern}`;
+
+            return `${globalCommandArgument}${excludeChar}${updatedPattern}`;
+        };
+
+        const toGlobPattern = pathToGlobPattern({
+            extensions: [],
+            cwd: rootFolder
+        });
+
+        const pattern = isGlob(inputPath) ? inputPath : toGlobPattern(inputPath);
```

Comment:
We can likely save computation by doing it directly when needed:

```suggestion
const pattern = isGlob(inputPath)
  ? inputPath
  : pathToGlobPattern({ extensions: [], cwd: rootFolder })(inputPath);
```

Reply:
Done, Thanks !